mods/cnc/maps/nod07c/nod07c.lua

```diff
@@ -0,0 +1,186 @@
+GDI3 = { units = { ['jeep'] = 1 }, waypoints = { waypoint0.Location, waypoint1.Location, waypoint3.Location, waypoint4.Location, waypoint3.Location, waypoint2.Location, waypoint5.Location, waypoint6.Location, waypoint2.Location, waypoint7.Location } }
+MTANK = { units = { ['mtnk'] = 1 }, waypoints = { waypoint14.Location, waypoint5.Location } }
+
+GDIEntrance = { Actor118, Actor135, Actor136, Actor141, Actor164, GDIBuilding3, GDIBuilding9, GDIBuilding10 }
+GunboatPatrolPath = { GunboatLeft.Location, GunboatRight.Location }
+
+targetsKilled = 0
+
+AutoGuard = function(guards)
+	Utils.Do(guards, function(guard)
+		Trigger.OnDamaged(guard, function(guard, attacker)
+			if not guard.IsDead and not attacker.IsDead then
+				guard.AttackMove(attacker.Location)
+			end
+		end)
+	end)
+end
+
+InsertNodUnits = function()
+	Media.PlaySpeechNotification(player, "Reinforce")
+	Reinforcements.Reinforce(player, { 'ltnk'}, { ReinforcementsTopSpawn.Location, ReinforcementsTankRally.Location }, 1)
+	local Engineers = Reinforcements.Reinforce(player, NodUnitsEngineers, { ReinforcementsTopSpawn.Location, ReinforcementsEngineersRally.Location }, 10)
+	Reinforcements.Reinforce(player, NodUnitsRockets, { ReinforcementsBottomSpawn.Location, ReinforcementsRocketsRally.Location }, 10)
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		Reinforcements.Reinforce(player, NodUnitsGunners, { ReinforcementsBottomSpawn.Location, ReinforcementsGunnersRally.Location }, 10)
+		Reinforcements.Reinforce(player, NodUnitsFlamers, { ReinforcementsTopSpawn.Location, ReinforcementsFlamersRally.Location }, 10)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local unitsA = Reinforcements.ReinforceWithTransport(player, 'tran.in', NodUnitsVehicles1, { GunboatRight.Location, ReinforcementsHelicopter1Rally.Location }, { GunboatRight.Location })[2]
+	
+		Trigger.AfterDelay(DateTime.Seconds(3), function()
+			local unitsB = Reinforcements.ReinforceWithTransport(player, 'tran.in', NodUnitsVehicles2, { GunboatRight.Location, ReinforcementsHelicopter2Rally.Location }, { GunboatRight.Location })[2]
+
+			Utils.Do(unitsB, function(unit)
+				unitsA[#unitsA + 1] = unit
+			end)
+
+			Trigger.OnAllKilled(unitsA, function()
+				if not defendersActive then
+					player.MarkFailedObjective(NodObjective3)
+				end
+			end)
+		end)
+	end)
+
+	Trigger.OnAllKilled(Engineers, function()
+		if not player.IsObjectiveCompleted(NodObjective1) then 
+			player.MarkFailedObjective(NodObjective1)
+		end
+	end)
+end
+
+SendGuards = function(team)
+	for type, amount in pairs(team.units) do
+		local count = 0
+		local actors = enemy.GetActorsByType(type)
+		Utils.Do(actors, function(actor)
+			if actor.IsIdle and count < amount then
+				actor.Patrol(team.waypoints, true, DateTime.Seconds(25))
+				count = count + 1
+			end
+		end)
+	end
+end
+
+SendDefenders = function(team)
+	defendersActive = true
+	player.MarkCompletedObjective(NodObjective3)
```

Comment:
It's a bit weird that you still get this granted even when you attack those units only after you're already in the base.